src/azul/terra.py

```diff
@@ -296,15 +299,20 @@ def _request(self,
         timeout = config.terra_client_timeout
         log.debug('_request(%r, %r, fields=%r, headers=%r, timeout=%r, body=%r)',
                   method, url, fields, headers, timeout, body)
-        response = self._http_client.request(method,
-                                             url,
-                                             fields=fields,
-                                             headers=headers,
-                                             # FIXME: Service should return 503 response when Terra client times out
-                                             #        https://github.com/DataBiosphere/azul/issues/3968
-                                             timeout=timeout,
-                                             retries=False,
-                                             body=body)
+        try:
+            response = self._http_client.request(method,
+                                                 url,
+                                                 fields=fields,
+                                                 headers=headers,
+                                                 # FIXME: Service should return 503 response when Terra client times out
+                                                 #        https://github.com/DataBiosphere/azul/issues/3968
+                                                 timeout=timeout,
+                                                 retries=False,
+                                                 body=body)
+        except ReadTimeoutError:
```

Comment:
Check the class hierarchy, but I think this is overly specific.

```suggestion
        except TimeoutError:
```

Reply:
```
>>> from urllib3.exceptions import ReadTimeoutError
>>> issubclass(ReadTimeoutError, TimeoutError)
False
```

And changing the `except` clause to `TimeoutError` results in the original exception propagating uncaught:

```
$ git diff
diff --git a/environment.py b/environment.py
index 78408019..5471732d 100644
--- a/environment.py
+++ b/environment.py
@@ -452,7 +452,7 @@ def env() -> Mapping[str, Optional[str]]:
         # configured, separated by a colon. The first is for time-sensitive
         # contexts such as API Gateway. The second is for contexts in which we
         # can afford to be more patient.
-        'AZUL_TERRA_TIMEOUT': '5:20',
+        'AZUL_TERRA_TIMEOUT': '0.1:0.1',
 
         # The URL of the Terra Data Repository instance to index metadata from.
         'AZUL_TDR_SERVICE_URL': None,
diff --git a/src/azul/terra.py b/src/azul/terra.py
index c4ec75e7..6c2f5705 100644
--- a/src/azul/terra.py
+++ b/src/azul/terra.py
@@ -309,7 +309,7 @@ class TerraClient(OAuth2Client):
                                                  timeout=timeout,
                                                  retries=False,
                                                  body=body)
-        except ReadTimeoutError:
+        except TimeoutError:
             raise RequirementError(f'Did not recieve response from {url!r} '
                                    f'within {timeout} seconds')
```

```
>>> from azul import config
>>> config.terra_client_timeout
0.1
>>> from azul.terra import TDRClient
>>> TDRClient.for_indexer().snapshot_names_by_id()
Traceback (most recent call last):
...
urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='jade.datarepo-dev.broadinstitute.org', port=443): Read timed out. (read timeout=0.1)
```

Reply:
Oh, unless you meant `urllib3.exceptions.TimeoutError` instead of the builtin `TimeoutError`. The former works fine.